### DISCUSSION TITLE
Setting select to nullable not working on save

### DISCUSSION THREAD

**Discussion by khanakia** [filamentphp/filament#7712] [Help]
I have a nullable column and select i have allowed nullable but when i remove select value and save the values comes back.

Check video please:

https://github.com/filamentphp/filament/assets/2174170/fbf8dbd0-dcaf-4314-b7aa-9d83a01d8e7c

**Comment by danharrin** [comment]
You didn't send any code so how can we help?

**Reply by khanakia** [reply]
Hello,

this is what i have as you can see state_id is nullable and when i clear the select field and saving does not works
Note these points i have checked already
1. I have all the fields are unguarded already
2. The table column state_id type is nullable
3. I debug the laravel query log it sends this only like i updated the rate and unselected the state_id `update "tax_rates" set "rate" = ?, "updated_at" = ? where "id" = ? ["35.001","2023-08-13 08:13:44",2] `

It should send `stated_id=null` also

Note: I am using Postgres as my database

```php
return $form
            ->schema([
                Forms\Components\Select::make('country_id')
                    ->searchable()
                    ->preload()
                    ->live()
                    ->relationship('country', 'name')
                    ->nullable(),
                Forms\Components\Select::make('state_id')
                    ->searchable()
                    ->preload()
                    ->relationship('state', 'name')
                    ->options(fn (Get $get): Collection => State::query()
                    ->where('country_id', $get('country_id'))
                    ->pluck('name', 'id'))
                    ->nullable(),
                Forms\Components\Select::make('tax_class_id')
                    ->relationship('tax_class', 'name')
                    ->required(),
                Forms\Components\Select::make('tax_rate_code_id')
                    ->relationship('tax_rate_code', 'name')
                    ->required(),
                Forms\Components\TextInput::make('rate')
                    ->required()
                    ->numeric(),
                Forms\Components\Toggle::make('status'),
            ]);
```

**Reply by khanakia** [reply]
I have created a demo reproducible repo also here https://github.com/khanakia/filament_selectnull

It contains very minimal code just to reproduce the issue in MYSQL also. 

### How to run
1. I have committed the `.env` and `filament_selectnulldb.sql` files within the root directory
2. Restore the mysql database file
3. Run `php artisan serve`
4. Login with user: `admin@admin.com` and pass: `admin`
5. Edit any Item and unselect the brand and save


Here's the attached video:

https://github.com/filamentphp/filament/assets/2174170/6b1a45b0-bba4-4443-bcf6-a5b7dbcc9f28

**Reply by khanakia** [reply]
@danharrin This bug only exists in the newer version `^3.0-stable (v3.0.19)`

I was testing the filament demo main repo and I just downgraded to the same version as demo repo `v3.0.14` to my repo and it worked.

**Reply by danharrin** [reply]
Please open an issue

**Comment by khanakia** [comment]
Closing in favor of this issue: https://github.com/filamentphp/filament/issues/7715
